### DISCUSSION TITLE
Pimcore Extension Development Best Practices

### DISCUSSION THREAD

**Discussion by 80982proton** [pimcore/pimcore#16289] [Q&A / Support]
I would like to develop a Pimcore extension. I have found some information, including https://github.com/pimcore/bundle-generator. My question is, what is the best way to develop a Pimcore extension that you might want to publish on the marketplace?

Does anyone know how the CoreShop extension, for example, was developed? This extension is much more than just a bundle!

**Comment by partenit** [comment]
https://pimcore.com/docs/platform/Pimcore/Extending_Pimcore/Bundle_Developers_Guide/
